app/services/cross_strategy_coordinator.py

```
@@ -0,0 +1,597 @@
+        try:
+            holdings = positions.get("holdings", [])
+            total_value = positions.get("total_value_usd", 1)
+            
+            symbol_value = 0
+            for holding in holdings:
+                if holding.get("symbol", "").upper() == symbol.upper():
+                    symbol_value += holding.get("value_usd", 0)
+            
+            return symbol_value / total_value if total_value > 0 else 0
+            
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)